harvester/harvest/alma.py

```diff
@@ -0,0 +1,192 @@
+
+        harvest_type_map = {"full": "full", "incremental": "daily"}
+        date_regex = re.compile(r".+?alma-(\d{4}-\d{2}-\d{2})-.*")
+
+        for filepath in filepaths:
+
+            # skip for run type
+            if harvest_type_map[self.harvest_type] not in filepath:
+                continue
+
+            # parse date from filepath
+            match = date_regex.match(filepath)
+            if not match:  # pragma: nocover
+                message = f"Could not parse date from filepath: {filepath}"
+                logger.warning(message)
+                continue
+            filepath_date = convert_to_utc(date_parser(match.group(1)))
+
+            # skip if date out of bounds for harvester from/until dates
+            if self.from_datetime_object and filepath_date < self.from_datetime_object:
+                continue
+            if self.until_datetime_object and filepath_date >= self.until_datetime_object:
+                continue
+
+            yield filepath
+
+    def _list_s3_xml_files(self) -> list[str]:
+        """Return a list of S3 URIs for extracted Alma XML files.
+
+        Example self.input_files = "s3://timdex-extract-dev-222053980223/alma/"
+        """
+        bucket, prefix = self.input_files.replace("s3://", "").split("/", 1)
+        s3_objects = S3Client.list_objects_uri_and_date(bucket, prefix)
+        return [
+            s3_object[0]
+            for s3_object in s3_objects
+            if s3_object[0].lower().endswith(".xml")
+        ]
+
+    def _list_local_xml_files(self) -> list[str]:
+        """Return a list of local filepaths of extracted Alma XML files."""
+        return glob.glob(f"{self.input_files.removesuffix('/')}/*.xml")
+
+    def parse_marc_records_from_files(self) -> Iterator[MARCRecord]:
+        """Yield parsed MARCRecords from list of filepaths."""
+        for filepath in self._list_xml_files():
+            with smart_open.open(filepath, "rb") as f:
+                context = etree.iterparse(f, events=("end",), tag="record")
+                for _event, elem in context:
+                    yield MARCRecord(elem)
+                    elem.clear()
+                    while elem.getprevious() is not None:
+                        del elem.getparent()[0]
+
+    def filter_geospatial_marc_records(
+        self, marc_records: Iterator[MARCRecord]
+    ) -> Iterator[MARCRecord]:
+        """Yield geospatial MARC records by filtering on defined criteria."""
+        for i, record in enumerate(marc_records):
+            if i % 10_000 == 0 and i > 0:  # pragma: nocover
+                message = f"{i} MARC records scanned for geospatial filtering"
+                logger.info(message)
+
+            # skip if leader doesn't have a/c/n/p
+            if record.leader[5] not in ("a", "c", "n", "p"):
+                continue
+
+            # skip if Genre/Form 655 does not contain "Maps."
+            if not any(
+                "Maps." in subfield.value
```

Comment:
Perhaps we should make compare the values in lowercase formats? For example, `alma-2023-12-31-daily-extracted-records-to-index_01.xml` contains the following records. 
<img width="423" alt="image" src="https://github.com/MITLibraries/geo-harvester/assets/29102054/4da6e0c8-f1c8-4236-bea3-a833ed9a567f">

It seems like _at least one_ entry will always contain "Maps." if it is a GIS record, but this change wouldn't hurt (I think)!

Maybe:

```
"maps" in subfield.value.rstrip(".").lower().split()
for form in record["655"]
for subfield in form.subfield("a")
```